### ISSUE TITLE
nix-shell fails with HTTP 404 for required package

### ISSUE THREAD

**Issue by acristoffers** [zigtools/zls#749]
I'm totally new to nix, so maybe it's something wrong in my end, but `nix-shell` fails with the following error:

```
building '/nix/store/dzw1wcyd051p8wp8aql8ljxkkj94d6zp-source.drv'...

trying https://github.com/arqv/zig-overlay/archive/5b9504b8bff072553051d6e130727f7f5c0715c3.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     9    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
error: builder for '/nix/store/dzw1wcyd051p8wp8aql8ljxkkj94d6zp-source.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://github.com/arqv/zig-overlay/archive/5b9504b8bff072553051d6e130727f7f5c0715c3.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     9    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
       For full logs, run 'nix log /nix/store/dzw1wcyd051p8wp8aql8ljxkkj94d6zp-source.drv'.
(use '--show-trace' to show detailed location information)
```

Trying to download the file directly results in `HTTP 404`.

**Comment by SuperAuguste** [comment]
We recently updated our [flake.lock](https://github.com/zigtools/zls/blob/master/flake.lock) and [flake.nix](https://github.com/zigtools/zls/blob/master/flake.nix) to handle this very issue. Not sure how nix works but maybe the zls package just hasn't been updated to reflect this change? I sadly have no control over this.

**Comment by acristoffers** [comment]
I changed the title because who fails is `nix-shell` not `nix build`. The later is actually working, probably because of the changes in the flake files, but the former tries to download a file from a github repo that does not exist. Anyway, I noticed this while using the build script as a way to learn some nix, so maybe using nix-shell in this context not even makes sense. I'll leave the issue opened so someone with more understanding of nix can take a look if the behaviour is actually correct, but for me it can be closed.
